subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultSelfResolvingDependency.java

```diff
@@ -27,6 +27,8 @@
 import java.io.File;
 import java.util.Set;
 
+// TODO why do we override hashCode() but not equals() here?
```

Comment:
That hashcode implementation makes no sense for this class, it'll always be a constant. The standard hashcode would be appropriate as there is no equality for SeflResolvingDependencies.

Reply:
You're absolutely right! Not sure how I missed that. 😳